server/src/main/java/org/elasticsearch/rest/action/admin/cluster/RestClusterRerouteAction.java

```diff
@@ -70,6 +77,14 @@ public boolean allowSystemIndexAccessByDefault() {
 
     @Override
     public RestChannelConsumer prepareRequest(final RestRequest request, final NodeClient client) throws IOException {
+        if (request.hasParam("metric") == false || Objects.equals(request.param("metric"), "none") == false) {
```

Comment:
Could we have a test that this does indeed emit the deprecation warnings we think it does?